types/airtable/index.d.ts

```diff
@@ -1,107 +1,131 @@
-            create(records: TFields[], opts?: { typecast: boolean }): Promise<Records<TFields>>;
-            update(...args: any[]): Promise<any>;
-            replace(...args: any[]): Promise<any>;
-            destroy(...args: any[]): Promise<any>;
-        }
-
-        interface SortParameter {
-          field: string;
-          direction?: 'asc' | 'desc';
-        }
-
-        interface SelectOptions {
-            fields?: string[];
-            filterByFormula?: string;
-            maxRecords?: number;
-            pageSize?: number;
-            sort?: SortParameter[];
-            view?: string;
-            cellFormat?: 'json' | 'string';
-            timeZone?: string;
-            userLocale?: string;
-        }
-
-        interface Query<TFields extends object> {
-            all(): Promise<Records<TFields>>;
-            firstPage(): Promise<Records<TFields>>;
-            eachPage(pageCallback: (records: Records<TFields>, next: () => void) => void): Promise<void>;
-        }
-
-        interface Record<TFields> {
-            id: string;
-            fields: TFields;
-        }
-
-        type Records<TFields> = ReadonlyArray<Record<TFields>>;
-
-        interface Attachment {
-            id: string;
-            url: string;
-            filename: string;
-            size: number;
-            type: string;
-            thumbnails?: {
-                small: Thumbnail;
-                large: Thumbnail;
-                full: Thumbnail;
-            };
-        }
-
-        interface Thumbnail {
-            url: string;
-            width: number;
-            height: number;
-        }
-
-        interface Collaborator {
-          id: string;
-          email: string;
-          name: string;
-        }
-    }
+// TypeScript Version: 3.7
+
+export type baseCreateFunctor = (tableName: string) => Table;
+
+export default class Airtable {
+    constructor(options?: AirtableOptions);
+    base(appId: string): baseCreateFunctor;
+    static base(appId: string): baseCreateFunctor;
+    static configure(options: AirtableOptions): void;
+}
+
+export class Error {
+    constructor(error: Error | string, message: string, statusCode: number | string);
+    toString(): string;
+}
+
+export interface FieldSet {
+    [key: string]:
+        | undefined
+        | string
+        | number
+        | boolean
+        | Collaborator
+        | ReadonlyArray<Collaborator>
+        | ReadonlyArray<string>
+        | ReadonlyArray<Attachment>;
+}
+
+export interface NewFieldData<TFields> {
+    fields: Partial<TFields>;
+}
+
+export interface ExistingFieldData<TFields> extends NewFieldData<TFields> {
+    id: string;
+}
+
+export interface AirtableOptions {
+    apiKey?: string;
+    endpointUrl?: string;
+    apiVersion?: string;
+    allowUnauthorizedSsl?: boolean;
+    noRetryIfRateLimited?: boolean;
+    requestTimeout?: number;
+}
+
+export class Base {
```

Comment:
Users aren't really meant to instantiate this themselves. Could we remove the export (or maybe mark the constructor as private, if TypeScript allows such a thing)?

The same comment applies to `Table` and `Record`.